temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,254 @@
+      Functions.Proc3<T, A1, A2> update, A1 arg1, A2 arg2) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2});
+  }
+
+  public static <T, A1, A2, A3> Builder<Void> newBuilder(
+      Functions.Proc4<T, A1, A2, A3> update, A1 arg1, A2 arg2, A3 arg3) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2, arg3});
+  }
+
+  public static <T, A1, A2, A3, A4> Builder<Void> newBuilder(
+      Functions.Proc5<T, A1, A2, A3, A4> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4});
+  }
+
+  public static <T, A1, A2, A3, A4, A5> Builder<Void> newBuilder(
+      Functions.Proc6<T, A1, A2, A3, A4, A5> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4, A5 arg5) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4, arg5});
+  }
+
+  public static <R> Builder<R> newBuilder(String update, Class<R> resultType, Object[] args) {
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setResultClass(resultType).setUpdateName(update), null, args);
+  }
+
+  private static <R> Builder<R> createBuilder(Object updateFunc, Object[] args) {
+    SerializedLambda meta = JavaLambdaUtils.toSerializedLambda(updateFunc);
+    if (meta == null || meta.getImplMethodKind() != MethodHandleInfo.REF_invokeInterface) {
+      throw new IllegalArgumentException("Invalid method reference: " + updateFunc);
+    }
+
+    String updateName = null;
+    String workflowType = null;
+    Class<R> resultClass = null;
+    String updateMethodName = meta.getImplMethodName();
+    try {
+      Class<?> workflowClass = Class.forName(meta.getImplClass().replace('/', '.'));
+      workflowType = workflowClass.getSimpleName();
+      Method[] allMethods = workflowClass.getDeclaredMethods();
+      for (Method method : allMethods) {
+        if (method.getName().equals(updateMethodName)) {
+          resultClass = (Class<R>) method.getReturnType();
+          UpdateMethod annotation = method.getAnnotation(UpdateMethod.class);
+          if (annotation == null) {
+            throw new IllegalArgumentException(
+                "Method '" + updateMethodName + "' does not have UpdateMethod annotation");
+          }
+          updateName = method.getName();
+          if (!annotation.name().isEmpty()) {
+            updateName = annotation.name();
+          }
+        }
+      }
+      if (resultClass == null) {
+        throw new IllegalArgumentException(
+            "Update method with name '" + updateMethodName + "' not found on: " + workflowClass);
+      }
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException("Invalid method reference", e);
+    }
+
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setUpdateName(updateName).setResultClass(resultClass),
+        workflowType,
+        args);
+  }
```

Comment:
This was born out of necessity. I could not find a way to do this where the typed stub is available; as that has no access to the workflow options.